Mathlib/NumberTheory/Padics/Harmonic.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2023 Koundinya Vajjha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Koundinya Vajjha
+-/
+
+import Mathlib.NumberTheory.Padics.PadicIntegers
+import Mathlib.Data.Int.Log
+/-!
+
+The nth Harmonic number is not an integer. We formalize the proof using
+2-adic valuations. This proof is due to Kürschák.
+
+Reference:
+https://kconrad.math.uconn.edu/blurbs/gradnumthy/padicharmonicsum.pdf
+
+-/
+
+open BigOperators
+
+/-- The nth-harmonic number defined as a finset sum of consecutive reciprocals. -/
+def harmonic : ℕ → ℚ := fun n => ∑ i in Finset.range n, 1 / (i + 1)
```

Comment:
I think usually `n⁻¹` is preferred over `1 / n` (analogous to how `-n` is the preferred over `0 - n`).
```suggestion
def harmonic : ℕ → ℚ := fun n => ∑ i in Finset.range n, (↑(i + 1))⁻¹
```

Reply:
But we also have: https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/PSeries.html#Real.tendsto_sum_range_one_div_nat_succ_atTop which does `1/(i+1)`.